e2e/pos_test.go

```diff
@@ -613,3 +615,171 @@ func TestPoS_StakeUnstakeWithinSameBlock(t *testing.T) {
 
 	validateValidatorSet(senderAddr, client, t, false, numDummyStakers)
 }
+
+func getSnapshot(
+	client ibftOp.IbftOperatorClient,
+	blockNum uint64,
+	ctx context.Context,
+) (*ibftOp.Snapshot, error) {
+	snapshot, snapshotErr := client.GetSnapshot(ctx, &ibftOp.SnapshotReq{
+		Latest: false,
+		Number: blockNum,
+	})
+
+	return snapshot, snapshotErr
+}
+
+func getNearestNextEpochBlock(blockNum uint64, epochSize uint64) uint64 {
+	if epochSize > blockNum {
+		return epochSize
+	}
+
+	nearest := blockNum + epochSize/2
+	nearest = nearest - (nearest % epochSize)
+
+	if nearest < blockNum {
+		nearest += epochSize
+	}
```

Comment:
i feel we can make this more readable.

do you think this is faster than performing mod block size to check if we're already there, and then doing integer division and multiplication by epoch size to deduce the next epoch block?

Reply:
Simplified it by a lot 👍 


8a2ee9c